### PR TITLE
Fix invalid dates such as leap year

### LINKED ISSUE #249: [PE-D][Tester D] addMeeting automatically edits illegal dates in February
When addMeeting is called with the illegal dates 29 Feb [non-leap year], or 30/31 Feb [any year] (pictured), the date is automatically changed to the latest valid date instead of giving an error.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: weesiau/ped#10

### PR DESCRIPTION
**Pull request by sikai00**
Solution:
Resolver style of STRICT was used. Since `ParserUtil#parseDate` is used for all dates, all dates have been fixed

Tests were also written to check for it. Manual testing was done.
`editMeeting` date
image

`addClient`birthday
image

`editClient` birthday
image

`addMeeting` date
image

Closes #249, #244

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddClientCommandParserTest.java [hunk 1]
```diff
@@ -122,4 +130,36 @@ public void parse_invalidValue_failure() {
                 + ADDRESS_DESC_BOB + PRODUCT_DESC_PRODUCT2 + PRODUCT_DESC_PRODUCT1,
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddClientCommand.MESSAGE_USAGE));
     }
+
+    @Test
+    public void parse_invalidBirthday_failure() {
+        // boundary value analysis
```
Comment (Thing1Thing2) [inline src/test/java/seedu/address/logic/parser/AddClientCommandParserTest.java:136]: perhaps you can mention the EP whose boundary you are taking

Comment (sikai00) [reply src/test/java/seedu/address/logic/parser/AddClientCommandParserTest.java:136]: thought the 1 jan and 31 dec etc was clear, but will make it clearer, thanks!

**Comment by sikai00** [comment]
Commands where dates are used (and invalid dates can be tested for):
`addClient ... bd/DATE`
`editClient ... bd/DATE`
`addMeeting ... d/DATE`
`editClient ... d/DATE`
